### PR TITLE
Protect: Use ToggleControl Component From @automattic/jetpack-components

### PR DESCRIPTION
**Pull request by nateweller**
Extracted from https://github.com/Automattic/jetpack/pull/40317 with adjustments.

## Proposed changes:

* Use the `ToggleControl` component from the `components` package, instead of Protect's custom `FormToggle` and checkbox implementations.
* Delete the `FormToggle` component from Protect.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Test the firewall setting toggles in the Jetpack Protect plugin.
* Review the firewall settings and footer on various screen sizes.

## Screenshots

Screenshot 2024-12-09 at 9 31 07 AM

Screenshot 2024-12-09 at 9 30 54 AM

### CODE REVIEW

File: projects/plugins/protect/src/js/routes/firewall/firewall-footer.jsx [hunk 1]
```diff
@@ -41,95 +42,71 @@ const StandaloneMode = () => {
...
-			</div>
+		<div className={ styles[ 'share-data' ] }>
+			<Title mb={ 0 }>{ __( 'Share data with Jetpack', 'jetpack-protect' ) }</Title>
+			<ToggleControl
+				className={ styles[ 'share-data-toggle' ] }
+				checked={ Boolean( settings.jetpack_waf_share_data ) }
+				onChange={ handleShareDataChange }
+				disabled={ isUpdating }
+				size="small"
+				label={ __( 'Share basic data with Jetpack', 'jetpack-protect' ) }
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/routes/firewall/firewall-footer.jsx]:
```suggestion
				label={ __( 'Share basic data', 'jetpack-protect' ) }
```
Now that we have a general title for both toggles, I suppose we can simplify the `label` text. Maybe even further to `Basic data` and `Detailed data`? Out of scope for this task, but I also wonder if the `help` copy could better describe the toggle functionality, we're basically just repeating the same thing for each with basic/detailed.

File: projects/plugins/protect/src/js/routes/firewall/firewall-footer.jsx [hunk 2]
```diff
@@ -41,95 +42,71 @@ const StandaloneMode = () => {
...
+				help={ __(
+					'Allow Jetpack to collect basic data from blocked requests to improve firewall protection and accuracy.',
+					'jetpack-protect'
+				) }
+			/>
+			<ToggleControl
+				className={ styles[ 'share-data-toggle' ] }
+				checked={ Boolean( settings.jetpack_waf_share_debug_data ) }
+				onChange={ handleShareDebugDataChange }
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/routes/firewall/firewall-footer.jsx:86]: Noticing that the toggle restrictions/counter-actions aren't applying anymore - for example, when both settings are untoggled, toggling `Share detailed data` should also enable `Share basic data`, and untoggling `Share basic` should also untoggle `Share detailed data` if its enabled, etc

File: projects/plugins/protect/src/js/routes/firewall/index.jsx [hunk 3]
```diff
@@ -520,7 +498,6 @@ const FirewallPage = () => {
 					{ ( jetpackWafIpAllowListEnabled || ipAllowListHasContent ) && (
 						<div className={ styles[ 'manual-rules-section' ] }>
 							<Textarea
-								id="jetpack_waf_ip_allow_list"
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/routes/firewall/index.jsx]: Same as in the above, `id` is required to make the field functional.

File: projects/plugins/protect/src/js/routes/firewall/index.jsx [hunk 4]
```diff
@@ -464,7 +444,6 @@ const FirewallPage = () => {
 				{ ( ipBlockListEnabled || ipBlockListHasContent ) && (
 					<div className={ styles[ 'manual-rules-section' ] }>
 						<Textarea
-							id="jetpack_waf_ip_block_list"
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/routes/firewall/index.jsx]: Removing this `id` prop from the `Textarea` makes the section not possible to modify.

**Review by dkmyta — CHANGES_REQUESTED** [review]
Nice! Thanks for separating this out into a dedicated PR. Works great for the most part, left a few comments.

Nit - Wondering if using a `Text` component for the `label`, `help` prop content and/or manually updating styles (size, color, weight) to align more with other Protect sections might be better? Looks like the `help` content currently defaults to a random gray. Maybe something like:  
Screen Shot on 2024-12-09 at 09-53-32

When the footer section wraps, we should ensure there is a reasonable gap between the two sections: 
Screen Shot on 2024-12-09 at 09-58-38

Unrelated to your changes, but also noticing the `Trusted IP addresses` section gets a little tight in mobile view, `flex-wrap: wrap` could address that nicely: 
Screen Shot on 2024-12-09 at 10-00-40

File: projects/plugins/protect/src/js/routes/firewall/firewall-footer.jsx (hunk 2 above)
Comment (dkmyta) [reply projects/plugins/protect/src/js/routes/firewall/firewall-footer.jsx:86]: **UPDATE:** Settings are actually updated correctly, the UI just isn't automatically recognizing the changes 🤔

Comment (nateweller) [reply projects/plugins/protect/src/js/routes/firewall/firewall-footer.jsx:86]: Good catch, fixed this via https://github.com/Automattic/jetpack/pull/40510/commits/2d909bfa5d1ef59c06ed00638fc0d6b2c6f405c8 :)

File: projects/plugins/protect/src/js/routes/firewall/index.jsx (hunk 4 above)
Comment (nateweller) [reply projects/plugins/protect/src/js/routes/firewall/index.jsx]: Restored!

File: projects/plugins/protect/src/js/routes/firewall/index.jsx (hunk 3 above)
Comment (nateweller) [reply projects/plugins/protect/src/js/routes/firewall/index.jsx]: Restored!

**Comment by nateweller** [comment]
@dkmyta Updated the styles accordingly!

Desktop:

Screenshot 2024-12-09 at 11 41 50 AM

Mobile:

Screenshot 2024-12-09 at 11 42 02 AM

**Review by dkmyta — APPROVED** [review]
LGTM!

**Pull request merged**